crates/uv-client/src/base_client.rs

```diff
@@ -127,7 +127,18 @@ impl BaseClientBuilder<'_> {
             allow_insecure_host: vec![],
             native_tls: false,
             connectivity: Connectivity::Online,
-            retries: DEFAULT_RETRIES,
+            retries: env::var_os(EnvVars::UV_HTTP_RETRIES)
+                .and_then(|value| {
+                    if let Some(value) = value.to_str() {
+                        value.parse::<u32>().inspect_err(|err| 
+                            warn_user_once!("`UV_HTTP_RETRIES` ({value:?}) cannot be parsed as a u32 and will be ignored: {err}")
+                        ).ok()
+                    } else {
+                        warn_user_once!("`UV_HTTP_RETRIES` ({value:?}) is not valid UTF-8 and will be ignored");
```

Comment:
Should this be a fatal error?

Reply:
Plausibly yeah, but I was hesitant to change the type in the PoC